crypto/x509/x509_vfy.c

```diff
@@ -494,14 +494,17 @@ int X509_verify_cert(X509_STORE_CTX *ctx) {
 
 static X509 *find_issuer(X509_STORE_CTX *ctx, STACK_OF(X509) *sk, X509 *x) {
   size_t i;
-  X509 *issuer;
+  X509 *issuer, *candidate = NULL;
   for (i = 0; i < sk_X509_num(sk); i++) {
     issuer = sk_X509_value(sk, i);
     if (ctx->check_issued(ctx, x, issuer)) {
-      return issuer;
+      candidate = issuer;
+      if (x509_check_cert_time(ctx, candidate, /*suppress_error*/1)) {
+        break;
+      }
```

Comment:
With expired certs, we might iterate over multiple candidates. When we change candidates, don't we need a call to "free" the previous?

Reply:
I think `candidate` is assigned the pointer within the `sk` X509 stack. These return pointers within the stack without additional reference bumps.
https://github.com/aws/aws-lc/blob/fbe7a0646d707016d661c2471ad82908404c0d19/crypto/stack/stack.c#L128

If we try freeing the assigned pointer, the original `X509` index pointer within `sk` will be unintentionally freed as well. Because of this behavior, users of the `find_issuer` function have to bump the reference each time the call is successful.
* https://github.com/aws/aws-lc/blob/fbe7a0646d707016d661c2471ad82908404c0d19/crypto/x509/x509_vfy.c#L260-L267
* https://github.com/aws/aws-lc/blob/fbe7a0646d707016d661c2471ad82908404c0d19/crypto/x509/x509_vfy.c#L529-L531

This is pretty confusing as well, I'll add a comment above `find_issuer` to clarify this usage behavior.